### PR TITLE
[WiP][CR]Sectorless overmap generation

### PR DESCRIPTION
**Pull request by Coolthulhu**
tl;dr Overmap generation that isn't limited to 144 specials per overmap

Initial work on new overmap generation system to replace the old sectioned one.

So far there is:
* Brute forcing something similar to Voronoi diagram of city distances, except with smaller distances because cities are spheres. For city count below 10 it's fine to naive-brute it, above that we'd need something more. Or we could just cap city count at sensible number.
* Making a map of oter_id to locations it "supports". That is, a "wilderness" location has forests and fields, so fields and forests map to wilderness, but fields also map to fields and forests to forests etc.
* Map of points that support a location
* Placing things on those supporting locations, but no way to look up if a thing will be supported yet

This is a workable, but not good overmap gen. The maps have most of the specials, lacking just the road stops, because road stops need existing roads and those are hard to get by randomly poking around.
For some weird reason road stops are mandatory.

Other problems:
* Overmap specials are badly configured, because it looks like someone either tried to force "my features are too cool not to appear, it's better if others' features don't appear" or simply didn't understand how mapgen works. For example, there can be up to 30 anthills and up to 500 spider pits. This isn't Australia. New generator actually WILL place those.
* City distance limits aren't enforced yet
* No options to enable this new feature yet
* Ugly code because I exposed internals and ids. Help with cleaning it up appreciated.
* Support for overmap id blacklist is necessary for "land" location, but it isn't easy here. Can I drop it and just change it to a whitelist?

### CODE REVIEW

**Comment by DangerNoodle** [comment]
This will be good to see.

**Comment by ZhilkinSerg** [comment]
> 30 anthills and up to 500 spider pits
I've always wondered what are such big numbers for

**Comment by vache** [comment]
Any thoughts on getting specials to spawn in cities?

**Comment by Coolthulhu** [comment]
This is bitrotten beyond recovery and would need a rewrite anyway.
